src/app/application.cpp

```diff
@@ -562,8 +562,8 @@ void Application::processParams(const QStringList &params)
         // skipTorrentDialog is false, meaning that the application setting
         // should be overridden.
         const bool showDialogForThisTorrent =
-            ((AddNewTorrentDialog::isEnabled() && skipTorrentDialog == TriStateBool::Undefined)
-             || skipTorrentDialog == TriStateBool::False);
+            ((AddNewTorrentDialog::isEnabled() && !skipTorrentDialog.has_value())
+             || !*skipTorrentDialog);
```

Comment:
what if `skipTorrentDialog` does not contain a value?

Reply:
Good catch 👍
Will fix it.